glib/src/gstring.rs

```diff
@@ -102,6 +102,67 @@ impl GStr {
     pub fn as_c_str(&self) -> &CStr {
         unsafe { CStr::from_bytes_with_nul_unchecked(self.to_bytes_with_nul()) }
     }
+
+    #[doc(alias = "g_utf8_collate")]
+    #[inline]
+    pub fn collate(&self, other: &Self) -> Ordering {
+        unsafe { ffi::g_utf8_collate(self.as_ptr(), other.as_ptr()) }.cmp(&0)
+    }
+
+    #[doc(alias = "g_utf8_casefold")]
+    #[inline]
+    pub fn casefold(&self) -> GString {
+        let str = unsafe { ffi::g_utf8_casefold(self.as_ptr(), self.len() as isize) };
+
+        unsafe { from_glib_full(str) }
+    }
+
+    #[doc(alias = "g_utf8_normalize")]
+    pub fn normalize(&self, mode: NormalizeMode) -> GString {
+        let str =
+            unsafe { ffi::g_utf8_normalize(self.as_ptr(), self.len() as isize, mode.into_glib()) };
+
+        unsafe { from_glib_full(str) }
+    }
+
+    #[doc(alias = "g_utf8_to_ucs4_fast")]
+    pub fn encode_utf32(&self) -> Slice<u32> {
+        let mut items_written = 0;
+
+        // SAFETY: GStr must be valid UTF-8
+        let str_as_utf32 =
+            unsafe { ffi::g_utf8_to_ucs4_fast(self.as_ptr(), self.len() as _, &mut items_written) };
+        unsafe { Slice::from_glib_full_num(str_as_utf32.cast(), items_written as usize) }
+    }
+}
+
+#[doc(alias = "GNormalizeMode")]
+pub enum NormalizeMode {
+    #[doc(alias = "G_NORMALIZE_DEFAULT")]
```

Comment:
this should be auto generated, add it to the Gir.toml file & run the generator script